VContainer/Assets/VContainer/Runtime/Registry.cs

```diff
@@ -109,12 +109,30 @@ public bool TryGet(Type interfaceType, out Registration registration)
             {
                 var openGenericType = RuntimeTypeCache.OpenGenericTypeOf(interfaceType);
                 var typeParameters = RuntimeTypeCache.GenericTypeParametersOf(interfaceType);
-                return TryFallbackToSingleElementCollection(interfaceType, openGenericType, typeParameters, out registration) ||
+                return TryGetClosedGenericRegistration(interfaceType, openGenericType, typeParameters, out registration) ||
+                       TryFallbackToSingleElementCollection(interfaceType, openGenericType, typeParameters, out registration) ||
                        TryFallbackToContainerLocal(interfaceType, openGenericType, typeParameters, out registration);
             }
             return false;
         }
 
+        bool TryGetClosedGenericRegistration(Type interfaceType, Type openGenericType,
+            Type[] typeParameters,
+            out Registration registration)
+        {
+            if (hashTable.TryGet(openGenericType, out var openGenericRegistration))
+            {
+                if (openGenericRegistration.Provider is OpenGenericInstanceProvider implementationRegistration)
+                {
+                    registration = implementationRegistration.GetClosedRegistration(interfaceType, typeParameters);
+                    return true;
+                }
+            }
+
+            registration = null;
+            return false;
+        }
+
         public bool Exists(Type type) => hashTable.TryGet(type, out _);
```

Comment:
```suggestion
        public bool Exists(Type type)
        {
            if (type.IsConstructedGenericType)
            {
                type = RuntimeTypeCache.OpenGenericTypeOf(type);
            }
            return hashTable.TryGet(type, out _);
        } 
```

I tested out your implementation in my project and ran into an issue here.
Exists doesn't behave correctly and returns false when checking registered open generics.

Reply:
Thanks! I forgot the scopes tests and missed that error. I'll add them and your fix as soon as the compilation errors in the master branch are fixed

Reply:
When I resolve IEnumerable<OpenGenericInterface<string>> or IReadOnlyList<OpenGenericInterface<string>>, I get an empty list. I expect to return me a list for all classes that are registered as this open generic interface. 

Additionally in regard to that, I suggest to add tests for resolving multiple instances of an interface.